sigmastate/src/main/scala/sigmastate/interpreter/Hint.scala

```diff
@@ -0,0 +1,109 @@
+package sigmastate.interpreter
+
+import java.math.BigInteger
+
+import sigmastate.UncheckedTree
+import sigmastate.Values.SigmaBoolean
+import sigmastate.basics.FirstProverMessage
+import sigmastate.basics.VerifierMessage.Challenge
+
+/**
+  * A hint for a prover which helps the prover to prove a statement. For example, if the statement is "pk1 && pk2",
+  * and the prover knows only a secret for the public key pk1, the prover fails on proving without a hint. But if the
+  * prover knows that pk2 is known to another party, the prover may prove the statement (with an empty proof for "pk2").
+  */
+trait Hint
+
+/**
+  * A hint which is indicating that a secret associated with its public image "image" is already proven.
```

Comment:
Seems to be incorrect as below, `SimulatedSecretProof` also extends `SecretProven` but since its a simulated secret proof, it means that secret should not be known,